com.reprezen.swagedit/src/com/reprezen/swagedit/preferences/SwaggerValidationPreferences.java

```diff
@@ -31,6 +33,7 @@
         implements IWorkbenchPreferencePage, IPropertyChangeListener {
 
     public SwaggerValidationPreferences() {
+        // GRID is needed because we are not attaching the editor fields directly to FieldEditorParent, but to its child
```

Comment:
This does not make the group work properly in FieldEditorPreferencePage. The problems are much deeper than that. It works properly in certain conditions, but not all. Adding another group or fields with different column requirements will likely lead to unfixable format issues, which is why I'd recommend a comment to this effect here. This breaks the assumption of the FEPP class.

The eclipse documentation on FEPP is misleading on this. It does not point out the risks of using your own layouts, and in fact the FEPP javadoc invites you to override #createLayout if you have special layout needs. This article is closer to correct: http://www.eclipse.org/articles/Article-Field-Editors/field_editors.html. Still, it makes it sound like you can use your own field editors at liberty within your own custom layout. Even that's not strictly true. The bottom line is that FieldEditors all manipulate the containing grid's layout, in ways that cannot be managed properly. So you can use FieldEditors within your own layout, as long as that layout puts every group of "sibling" field editors under the control of the same GridLayout, with nothing else in that GridLayout.